test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs

```diff
@@ -1940,13 +1941,19 @@ public async Task RestoreCommand_RestoreExactVersionWithFailingSourceAsync()
                     LockFilePath = Path.Combine(projectDir, "project.lock.json")
                 };
 
-                var command = new RestoreCommand(request);
-
                 // Act
-                var result = await command.ExecuteAsync();
+                var command = new RestoreCommand(request);
+                RestoreResult result = await command.ExecuteAsync();
+                await result.CommitAsync(logger, CancellationToken.None);
 
                 // Assert
-                Assert.True(result.Success);
+                Assert.False(result.Success);
+                IAssetsLogMessage[] logMessages = result.LogMessages.ToArray();
+                Assert.Equal(4, logMessages.Count());
+                Assert.Equal(NuGetLogCode.NU1300, logMessages[0].Code);
```

Comment:
How many sources are failing here? 
Can the messages be merged somewhere?

Reply:
1 source is failing.
yes, i'll investigate.

Reply:
working on fix. looks promising.